cuckoo/cuckoo_result.py

```diff
@@ -1727,6 +1723,73 @@ def _remove_safe_roots(process_tree: List[Dict[str, Any]], process_tree_hashes:
             process_tree.pop(index - num_removed)
             num_removed += 1
 
+def _create_hashed_leaf(parent: str, node: Dict[str, Any], hashes: List[str]) -> None:
+    # Extract the two pieces of info needed from the node
+    image = node["image"]
+    children = node["children"]
+
+    info = parent + image
+
+    encoded = info.encode()
+    # hexdigest() returns the SHA256 encoded string of the object, encoded256
+    encoded256 = sha256(encoded).hexdigest()
+
+    if not children:
+        hashes.append(encoded256)
+    for child in children:
+        _create_hashed_leaf(encoded256, child, hashes)
+
+def _create_leaf_hashes(process_tree: List[Dict[str, Any]]) -> List[List[str]]:
```

Comment:
Same here, the `_create_leaf_hashes` method contains a lot of similar code to the `create_process_tree_hashes` method. Deduplicate the code by combining them